drivers/sensor/mma8451q/mma8451q.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2018 Lars Knudsen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <i2c.h>
+#include <kernel.h>
+#include <misc/util.h>
+#include <sensor.h>
+
+#include "mma8451q.h"
+
+/* Data-sheet: https://www.nxp.com/docs/en/data-sheet/MMA8451Q.pdf */
+
+#define OUT_X_MSB     0x01
+#define OUT_Y_MSB     0x03
+#define OUT_Z_MSB     0x05
+
+#define XYZ_DATA_CFG  0x0E
+
+#define CTRL_REG1     0x2A
+#define CTRL_REG2     0x2B
+#define CTRL_REG3     0x2C
+#define CTRL_REG4     0x2D
+#define CTRL_REG5     0x2E
+
+#define RANGE_2G      0x00
+#define RANGE_4G      0x01
+#define RANGE_8G      0x02
+
+static int mma8451q_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	struct mma8451q_data *drv_data = dev->driver_data;
+
+	u8_t buf[6];
+
+	/* Read data from all three axes at the same time. */
+	if (i2c_burst_read(drv_data->i2c, CONFIG_MMA8451Q_I2C_ADDR,
+			   OUT_X_MSB, buf, 6) < 0) {
+		SYS_LOG_ERR("Could not read accelerometer data");
+		return -EIO;
+	}
+
+	drv_data->x = (buf[0] * 256) + buf[1];
+	drv_data->y = (buf[2] * 256) + buf[3];
+	drv_data->z = (buf[4] * 256) + buf[5];
+
+	return 0;
+}
+
+static int mma8451q_channel_get(struct device *dev,
+				   enum sensor_channel chan,
+				   struct sensor_value *val)
+{
+	struct mma8451q_data *drv_data = dev->driver_data;
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+		val->val1 = drv_data->x;
+		val->val2 = 0;
+		break;
+
+	case SENSOR_CHAN_ACCEL_Y:
+		val->val1 = drv_data->y;
+		val->val2 = 0;
+		break;
+
+	case SENSOR_CHAN_ACCEL_Z:
+		val->val1 = drv_data->z;
+		val->val2 = 0;
+		break;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api mma8451q_driver_api = {
+	.sample_fetch = mma8451q_sample_fetch,
+	.channel_get = mma8451q_channel_get,
+};
+
+int mma8451q_init(struct device *dev)
+{
+	struct mma8451q_data *drv_data = dev->driver_data;
+
+	drv_data->i2c = device_get_binding(CONFIG_MMA8451Q_I2C_MASTER_DEV_NAME);
```

Comment:
Store the i2c dev name in the device config structure, just like the i2c address.